### PR TITLE
Write Test Case For Azure cli Action

### PR DESCRIPTION
**Pull request by PankajGovindRao**
Test Case For Azure cli Action

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -21,15 +22,18 @@ const run = async () => {
         }
 
         let inlineScript: string = core.getInput('inlineScript', { required: true });
-        let azcliversion: string = core.getInput('azcliversion', { required: true }).trim().toLowerCase();
-
+        let azcliversion: string = core.getInput('azcliversion', { required: true });
```
Comment (t-dedah) [inline src/main.ts:25]: Add `.trim().toLowerCase()` back

File: src/main.ts [hunk 2]
```diff
@@ -21,15 +22,18 @@ const run = async () => {
         }
 
         let inlineScript: string = core.getInput('inlineScript', { required: true });
-        let azcliversion: string = core.getInput('azcliversion', { required: true }).trim().toLowerCase();
-
+        let azcliversion: string = core.getInput('azcliversion', { required: true });
+       
         if (!(await checkIfValidCLIVersion(azcliversion))) {
+            console.log("INVALID VERSION")
```
Comment (t-dedah) [inline src/main.ts:28]: remove this console.log

File: src/main.ts (hunk 1 above)
Comment (PankajGovindRao) [reply src/main.ts]: Added  .trim().toLowerCase()

File: src/main.ts (hunk 2 above)
Comment (PankajGovindRao) [reply src/main.ts]: done

File: src/main.ts [hunk 3]
```diff
@@ -21,15 +22,18 @@ const run = async () => {
...
         let inlineScript: string = core.getInput('inlineScript', { required: true });
-        let azcliversion: string = core.getInput('azcliversion', { required: true }).trim().toLowerCase();
-
+        let azcliversion: string = core.getInput('azcliversion', { required: true });
+       
         if (!(await checkIfValidCLIVersion(azcliversion))) {
+            console.log("INVALID VERSION")
             core.setFailed('Please enter a valid azure cli version. \nSee available versions: https://github.com/Azure/azure-cli/releases.');
+            throw new Error('Please enter a valid azure cli version. \nSee available versions: https://github.com/Azure/azure-cli/releases.')
             return;
```
[the remark this answers is not in the document]
Comment (PankajGovindRao) [reply src/main.ts]: done

File: src/main.ts [hunk 4]
```diff
@@ -21,15 +22,18 @@ const run = async () => {
...
         }
 
         if (!inlineScript.trim()) {
             core.setFailed('Please enter a valid script.');
+            throw new Error('Please enter a valid script.')
             return;
```
[the remark this answers is not in the document]
Comment (PankajGovindRao) [reply src/main.ts]: done

File: .github/workflows/ci-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+name: ci-workflow
+on:
```
Comment (t-dedah) [inline .github/workflows/ci-workflow.yml:2]: Replace on with this - 

```
on:
  push:
  schedule:
    - cron: '0 */3 * * *'
```

File: .github/workflows/ci-workflow.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          EXPECTED_TO: fail
+        run: ts-node test/main.test.ts
+        
+      - name: Inline Script Test - Negative
+        env:
+          INPUT_AZCLIVERSION: 2.0.72
+          INPUT_INLINESCRIPT: " "
+          EXPECTED_TO: fail
+        run: ts-node test/main.test.ts
+
```
Comment (t-dedah) [inline .github/workflows/ci-workflow.yml:47]: Add post status action at the end

```
 - name: Post to slack on failure
        if: failure()
        uses: 8398a7/action-slack@v3.9.1
        with:
            status: ${{ job.status }}
            fields: repo,message,author,action,ref,workflow
        env:
            SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

File: .github/workflows/ci-workflow.yml (hunk 5 above)
Comment (PankajGovindRao) [reply .github/workflows/ci-workflow.yml:2]: Done

File: .github/workflows/ci-workflow.yml (hunk 6 above)
Comment (PankajGovindRao) [reply .github/workflows/ci-workflow.yml:58]: done

**Review by t-dedah — APPROVED** [review]

**Pull request merged**